src/Microsoft.AspNetCore.Mvc.TagHelpers/ScriptTagHelper.cs

```diff
@@ -337,6 +325,25 @@ private void BuildFallbackBlock(TagHelperAttributeList attributes, DefaultTagHel
             }
         }
 
+        private void AppendEncodedVersionedSrc(
+            string srcName,
+            string srcValue,
+            TagHelperContent builder,
+            bool javaScriptSrc = true)
+        {
+            if (AppendVersion == true)
+            {
+                srcValue = _fileVersionProvider.AddFileVersionToPath(srcValue);
+            }
+
+            if (javaScriptSrc)
```

Comment:
This should not be conditional. Appending the version could add characters that must be encoded.